### ISSUE TITLE
[JIT] Emitted bounds check comparing a constant index against a constant length

### ISSUE THREAD

**Issue by Zhentar** [dotnet/runtime#10686]
From dotnet/runtime#10685, the JIT mostly does an impressively good job of optimizing this code:
```cs
	public class SpanTest
	{
		private struct Data { public int a, b, c, d; }
		private static Data _data = default;

		public void IncrementA() => IncrementWrapper(MemoryMarshal.CreateSpan(ref _data.a, 4));
		[MethodImpl(MethodImplOptions.AggressiveInlining)]
		public static void IncrementWrapper(Span<int> span)
		{
			if ((uint)span.Length < 4) throw new IndexOutOfRangeException();
			Increment(ref span[0]);
		}
		private static void Increment(ref int val) => val++;
	}
```

```asm
TestHarness.SpanTest.IncrementA()
	mov     rcx,7FFBF8E206B0h
	mov     edx,5
	call    coreclr!coreclr_shutdown_2+0x10560
	mov     rax,1A0E5407108h
	mov     rax,qword ptr [rax]
	add     rax,8
	mov     edx,4 ;<--------
	cmp     edx,0 ;<--------
	jbe     00007ffb`f8d627da
	inc     dword ptr [rax]
	add     rsp,28h
	ret
```

That... seems like a rather easy bounds check to cut out.

category:cq
theme:bounds-checks
skill-level:intermediate
cost:small

**Comment by SingleAccretion** [comment]
This has been fixed in #49271. The original reproduction now compiles to the following:
```asm
       mov      rax, 0xD1FFAB1E
       mov      rax, gword ptr [rax]
       add      rax, 8
       inc      dword ptr [rax]
G_M37850_IG03:
       ret
```

**Comment by SingleAccretion** [comment]
Closing as per the above comment.
